bin/deployables

```diff
@@ -83,600 +99,829 @@ region=${DEPLOY_AWS_REGION}
 
-    echo
-    echo Deploying ECS task
-    echo image: ${image}
-    echo subfamily: ${subfamily}
-    echo
+  echo
+  echo Deploying ECS task
+  echo image: ${image}
+  echo subfamily: ${subfamily}
+  echo
 
-    if [ -z "$subfamily" ] ; then
-        family="${DEPLOY_APP_NAME}"
-        service="${DEPLOY_APP_NAME}"
-    else
-        family="${DEPLOY_APP_NAME}-${subfamily}"
-        service="${DEPLOY_APP_NAME}-${subfamily}"
-    fi
+  if [ -z "$subfamily" ] ; then
+    family="${DEPLOY_APP_NAME}"
+  else
+    family="${DEPLOY_APP_NAME}-${subfamily}"
+  fi
+  service="${family}"
+
+  echo
+  echo "Remembering current task definition, family: ${family}"
+  echo
+
+  previous_version=$( \
+    aws ecs describe-services \
+    --profile ${DEPLOY_AWS_PROFILE} \
+    --cluster ${DEPLOY_ECS_CLUSTER_NAME} \
+    --services ${service} \
+    | $JQ '.services[0].taskDefinition' \
+    )
 
-    echo
-    echo "Remembering current task definition, family: ${family}"
-    echo
+  echo
+  echo "Registering new task definition, family: ${family}"
+  echo
+
+  task_def_template=${DEPLOY_TASK_DEF_TEMPLATE//-FAMILY/-$subfamily}
 
-    previous_version=$( \
-        aws ecs describe-services \
-            --profile ${DEPLOY_AWS_PROFILE} \
-            --cluster ${DEPLOY_ECS_CLUSTER_NAME} \
-            --services ${service} \
-            | $JQ '.services[0].taskDefinition' \
+  if [[ ! -f ${task_def_template} ]] ; then
+    echo "Unable to find task definition template, task_def_template: ${task_def_template}"
+    exit 1
+  fi
+
+  task_def=$( \
+    DEPLOY_IMAGE_NAME=${image} \
+    DEPLOY_IMAGE_TAG=${image_tag} \
+    DEPLOY_ECS_CLUSTER_NAME=${DEPLOY_ECS_CLUSTER_NAME} \
+    DEPLOY_SUBFAMILY=${subfamily} \
+    DEPLOY_AWS_REGION=${DEPLOY_AWS_REGION} \
+    DEPLOY_APP_NAME=${family} \
+    ${DEPLOY_ENVSUBST_COMMAND:-envsubst} < "${task_def_template}"
+  )
+
+  if [[ -n "${DEPLOY_YAHTZEE_VARS}" ]]; then
+    echo "Task Def before Yahtzee: $(echo $task_def | $JQ .)"
+
+    new_task_def=$( \
+      echo ${task_def} | \
+      curl -sL -d @- -H "cookie: auth_token=${DEPLOY_YAHTZEE_AUTH_TOKEN}" \
+      "https://${DEPLOY_YAHTZEE_DOMAIN}/services/${DEPLOY_ECS_CLUSTER_NAME}/${DEPLOY_APP_NAME}/vars/json"
     )
 
-    echo
-    echo "Registering new task definition, family: ${family}"
-    echo
+    task_def="${new_task_def}"
+  fi
 
-    task_def_template=${DEPLOY_TASK_DEF_TEMPLATE//-FAMILY/-$subfamily}
+  echo "Deploying Task Def: $(echo $task_def | $JQ .)"
+
+  if [[ -z "${DEPLOY_TASK_ROLE_ARN}" ]] ; then
+    revision_target=$( \
+      aws ecs register-task-definition \
+      --profile ${DEPLOY_AWS_PROFILE} \
+      --container-definitions "$task_def" \
+      --family "${family}" \
+      | $JQ '.taskDefinition.taskDefinitionArn' \
+      )
+  else
+    revision_target=$( \
+      aws ecs register-task-definition \
+      --profile ${DEPLOY_AWS_PROFILE} \
+      --task-role-arn "${DEPLOY_TASK_ROLE_ARN}" \
```

Comment:
These two calls are the same except for this one flag. Is there a way to consolidate them further to reduce some of this duplication?

Reply:
Totally, will do.